librz/bin/bfile.c

```diff
@@ -428,45 +428,47 @@ RZ_API RZ_OWN RzPVector /*<RzBinFileHash *>*/ *rz_bin_file_compute_hashes(RzBin
 	}
 	RzListIter *hash_algos_iter = bin->default_hashes ? rz_list_iterator(bin->default_hashes) : NULL;
 
-	if (hash_algos_iter) {
-		while (hash_algos_iter) {
-			char *algo = rz_list_iter_get(hash_algos_iter);
-			if (!rz_hash_cfg_configure(md, algo)) {
-				goto rz_bin_file_compute_hashes_bad;
-			}
-		}
-
-		if (!rz_hash_cfg_init(md)) {
+	if (!hash_algos_iter) {
+		RZ_LOG_ERROR("bin: default hashes: no default hashes configured");
+		goto rz_bin_file_compute_hashes_bad;
+	}
+	while (hash_algos_iter) {
```

Comment:
This loop is wrong because is an infinite loop... use the normal looping for RzList as shown below.

```c
RzListIter *iter = NULL;
const char *algo = NULL;
rz_list_foreach (bin->default_hashes, iter, algo) {
// call rz_hash_cfg_configure(md, algo) ...
}
```